### DISCUSSION TITLE
format of `institution_id`

### DISCUSSION THREAD

**Discussion by mccrayc** [WCRP-CORDEX/discuss#10] [General]
### institution_id

OURANOS

### institution

Ouranos Consortium, Montreal, Quebec, Canada

**Comment by jesusff** [comment]
Thank you for the input. Registration is not yet open, though. The repository is still under test and the CV needs to be polished. Registration opening will be announced soon.

In any case, this is a nice example raising a problem to automatically parse the address. We expected the last two comma-separated fields to be the city and country. The optional but quite common specification of the region makes it difficult to extract these fields. Would it be very non-standard to specify the address as:

Ouranos Consortium, Montreal (Quebec), Canada

i.e. with the region in parenthesis if present?
For example, in the US: 

National Center for Atmospheric Research, Boulder (CO), USA

**Comment by mccrayc** [comment]
Ah okay thank you, I can resubmit once registration is open if that's better. Regarding the formatting, I think parentheses for the region is a reasonable way to do it that is easily parsed. I do think it is a good idea to allow regions here regardless of the way chosen (Springfield, USA can mean many different places, for example).

Another question: are accents accepted in the 'institution' field? If so, we would use 'Ouranos Consortium, Montréal (Québec), Canada'

**Comment by jesusff** [comment]
Thank you. Definitely, we need a way to include regions in the address. No need to resubmit your registration request. We can just let this issue open until solved.

Regarding accented characters. In the `institution` field they could be acceptable, but we may run into unexpected problems in the future. This information can be used in unforeseen ways in the future, and may have rendering problems due to the charset encoding. Just for reference, in CMIP6, I didn't find any specific restriction on accented characters, but they were not used in [institution](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_institution_id.json)s, even if there are some words that should carry them (e.g. Météorologiques https://github.com/WCRP-CMIP/CMIP6_CVs/issues/115#issuecomment-262180525)

@larsbuntemeyer what do you think?

**Comment by larsbuntemeyer** [comment]
Hey, sorry for the late response! I'm not really an expert on encoding but I think we should follow the obviously unwritten convention of CMIP6 (there is no convention of attribute data types) to avoid these special characters. I played around with it a little and, techically, i think it should be no problem (from what i read, default string attributes have type of `NC_CHAR` or an array of it). However, we would like to gurarantee a compatibility with `cmor>=3.7` and i fear that this kind of things could break it (although it should work in principle) if we go beyond CMIP6 conventions. By the way, we have the same problem with *umlaute* in Germany, e.g., `"UHH":"Universitat Hamburg, Hamburg 20148, Germany"` should actually contain `"Universität"` but you never know how this could cause problems in a downstream application...

**Comment by larsbuntemeyer** [comment]
However @mccrayc i think you put a lot of effort into `xclim` to allow a french vocabulary of metadata. Maybe you have more experience with it? I think if this would help downstream applications like `xclim` i would be tempted to just go for it and allow it. We seem to have more requests on this coming in anyway...

**Comment by larsbuntemeyer** [comment]
Here is an example downstream application that stumbles on encoding: https://github.com/pangeo-forge/pangeo-forge-recipes/issues/586

**Comment by gnikulin** [comment]
I didn't think about  the accents in `institution`. [The CMIP6 registration](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/main/.github/Institution_registration_template.md) says nothing about the accents, only  "_no restriction on character set_" for `institution`.  As @jesusff  already mentioned there are no any accents in the  [CMIP6 institution id CV](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_institution_id.json). I would recommend to avoid them, just to be on the safe side as we never know about all downstream applications and potential problems. 

@durack1 and @taylor13 did you have any issues with the accents in CMIP6 ?

**Comment by durack1** [comment]
@gnikulin @larsbuntemeyer for CMIP6, I imposed a unicode wash, so that accents etc got dropped before being committed to the repository, see [this](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/main/src/writeJson.py#L1300-L1364) code block for an example.

There were major issues in mapping across to HTML if accents and other unusual characters crept in, so doing this wash early and keeping the repo UTF-8 was the simplest solution

**Comment by taylor13** [comment]
Maybe we should mention in the registration guidance that this will be done?

**Comment by mccrayc** [comment]
No worries about the accents, we are fine leaving them out if that's the standard to be followed. 

I have also updated our intitution_id to be uppercase (OURANOS) to be consistent with our CMIP5 ID.

**Comment by durack1** [comment]
> Here is an example downstream application that stumbles on encoding: [pangeo-forge/pangeo-forge-recipes#586](https://github.com/pangeo-forge/pangeo-forge-recipes/issues/586)

@larsbuntemeyer this is an issue with the published data, rather than the controlled vocabularies/registered content. I do not believe the ESGF publisher currently does a unicode wash before entries are wedged into the ESGF (currently SOLR) indexes, but this would be an interesting question to answer. @sashakames can you speak to publisher behaviours?

**Comment by gnikulin** [comment]
@mccrayc is OURANOS an acronym or simply a name of a consortium  ?

**Comment by sashakames** [comment]
> > Here is an example downstream application that stumbles on encoding: [pangeo-forge/pangeo-forge-recipes#586](https://github.com/pangeo-forge/pangeo-forge-recipes/issues/586)
> 
> @larsbuntemeyer this is an issue with the published data, rather than the controlled vocabularies/registered content. I do not believe the ESGF publisher currently does a unicode wash before entries are wedged into the ESGF (currently SOLR) indexes, but this would be an interesting question to answer. @sashakames can you speak to publisher behaviours?

There is no specific "wash" code in the publisher to reformat strings.  To clarify, this would be for "full name" strings that get extracted from GA for inclusion in the record, rather than "id" string.  The latest ESGF proposal would be to drop such strings from the published record to the index, so adding such code may become irrelevant.

I don't think they are an acronym but the Montreal institution named after Greek.

**Comment by aulemahal** [comment]
@gnikulin Ouranos is the name of the consortium, not an acronym. The idea of capital letters was to have an identical ID with CMIP5.

**Comment by gnikulin** [comment]
OURANOS has been registered so we can close this issue or need more discussions about the accents ?
